pkg/ocm/gates.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright (c) 2020 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ocm
+
+import (
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+)
+
+func (c *Client) ListStsGates() (stsVersionGates []*cmv1.VersionGate, err error) {
+
+	versionGates, err := c.ListAllOcpGates()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, gate := range versionGates {
+		if gate.STSOnly() {
+			stsVersionGates = append(stsVersionGates, gate)
+		}
+	}
+
+	return
+
+}
+
+func (c *Client) ListOcpGates() (stsVersionGates []*cmv1.VersionGate, err error) {
+
+	versionGates, err := c.ListAllOcpGates()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, gate := range versionGates {
+		if !gate.STSOnly() {
+			stsVersionGates = append(stsVersionGates, gate)
+		}
+	}
+
+	return
+
+}
+
+func (c *Client) ListAllOcpGates() (versionGates []*cmv1.VersionGate, err error) {
+
+	versionGatesRequest := c.ocm.ClustersMgmt().V1().VersionGates()
+
+	page := 1
+	size := 100
+
+	for {
+		response, err := versionGatesRequest.List().
+			Page(page).
+			Size(size).
+			Send()
```

Comment:
Does this API support filtering by version? If so, we should allow this, otherwise we'd need to fetch all gates for all versions then filter on the client.